nncf/common/pruning/operations.py

```diff
@@ -311,6 +313,22 @@ class ReshapePruningOp(BasePruningOp):
     def _is_flatten(node: NNCFNode) -> bool:
         return len(node.layer_attributes.output_shape) == 2
 
+    @staticmethod
+    def _is_not_mixing_dim(node: NNCFNode, graph) -> bool:
+        input_edges = graph.get_input_edges(node)
+        input_shape = input_edges[0].tensor_shape
+
+        output_edges = graph.get_output_edges(node)
+        if not output_edges:
+            return True
+        output_shape = output_edges[0].tensor_shape
```

Comment:
please use `node.layer_attributes` here

Reply:
The logic of input/output shape collection is inside NNCFGraph building already. output_shape is required attribute of `ReshapeLayerAttributes`

Reply:
I know. But as you seen, we need to check whether `output_edges` are existed or not. There is the case that `output_edges` are not existed but `output_shape` is existed. In this case, we need to return True.

Reply:
You can check that layer attributes is existed, thus input_shape and output_shape are collected. `input_shapes` and `output_shapes` are required attributes of `ReshapeLayerAttributes`

Reply:
In some case, `output_shapes` are existed but `output_edges` are not existed. `tests/common/pruning/test_pruning_operations.py::test_reshape_metatype_mask_prop` is the one of the case. If we did not check whether `output_edges` are existed or not, and just use `output_shapes` of `ReshapeLayerAttributes`, we will return False in here, and the test could be failed. So we need to check `output_edges` are existed or not in here. If you do not want to use `output_shapes` like this, I will change this like below. 
```
        output_edges = graph.get_output_edges(node)
        if not output_edges:
            return True
        output_shape = node.layer_attributes.output_shape
```